### DISCUSSION TITLE
Wiring Raspberry Pi Zero 2 W with Pomoroni Plasma 2040?

### DISCUSSION THREAD

**Discussion by Txustadecosto** [awawa-dev/HyperSPI#47] [Q&A]
Hello everyone,
I need some help please.

The following setup:
-Rasperry Pi Zero 2W (Note: the Pins colours are Reversed, i soldered wrong, but doesent matter)
-Pomoroni Plasma 2040

Now I wanted to connect both via SPI because all USBs are occupied.
But I'm not sure whether I connected it correctly and whether the correct firmware is on the Plasma 2040.

I couldn't find anywhere how to connect these two components, only with other models.
Could you please tell me whether this is correct and whether I should pay attention to anything?
Maybe ist Possible to use an USB Hub instead to connect all togehther? But i read SPI are better and faster...

![20240526_190922](https://github.com/awawa-dev/HyperSPI/assets/141459714/02e106c7-533f-4c32-baec-997ca8da2743)

Thank you!

**Comment by awawa-dev** [comment]
Hi
First I try to understand this scheme https://cdn.shopify.com/s/files/1/0174/1800/files/plasma2040_schematic.pdf?v=1629994431 to find out which GPIO are used for SDA SCL INT pins on the board. It's not clear to me, maybe I will need to perform some experiments to figure it out.

**Reply by Txustadecosto** [reply]
Hi, thx for the fast reply. I try for hours to find a solution. Even the Schematic don't help me.
Maybe it's not possible? Do you think I have to buy a Raspberry 4 to conect the Plasma 2040 over USB A?
I try to copied you master/Slave Raspberrypi/Pico project from you post, and apply here, but vas not sure for the INT pin on Plasma.

**Reply by Txustadecosto** [reply]
It's possible to use an USB hub to conect The Zero + Grabber + Plasma together?

**Reply by awawa-dev** [reply]
Yes, it is also possible to connect Plasma (with HyperSerialPico) to Zero 2w via hub.

**Reply by awawa-dev** [reply]
it is possible to identify these GPIOs one by one by turning them on high and using a multimeter on pins to check whether there is 3.3v there. But I won't do it straight away, I'd have to dig out this setup.

**Reply by awawa-dev** [reply]
Maybe it goes this way?
![obraz](https://github.com/awawa-dev/HyperSPI/assets/69086569/7a03c85e-95cf-4fd4-b8af-937ac303bbae)
![obraz](https://github.com/awawa-dev/HyperSPI/assets/69086569/56bd2cb3-0082-4018-b401-ae511e0afb3f)

if so this board doesnt expose SPI unfortunately:
```
rp2040 allows hardware SPI on corresponding pairs of pins:
spi0 ⇒ Data/Clock: GPIO3/GPIO2, GPIO19/GPIO18, GPIO7/GPIO6
spi1 ⇒ Data/Clock: GPIO11/GPIO10, GPI15/GPIO14, GPIO27/GPI26
```
they connected GPIO18 to the LED on board :facepalm:

**Reply by Txustadecosto** [reply]
If so, that's so sad not able to conect via SPI.

I can't test the complete system because the Strips don't arrive yet, hope at the end if the week.
I use the USB hub now, but I have another problem:
The Zero Crashes sometimes. I think it's the grabber (MS2131) because I get the same problem with a pi5, only difference, the Pi5 starts again, the Zero stays online until On/off.

**Reply by awawa-dev** [reply]
Maybe it's a power supply problem? ms2131 requires a lot of power, more than older ms2109. Check `dmesg` logs!. You can stop HyperHDR service and run HyperHDR from a command line to get more logs when it crashes: https://github.com/awawa-dev/HyperHDR/wiki/Installation#linux
